homeassistant/components/mqtt/__init__.py

```diff
@@ -1104,7 +1104,7 @@ def availability_message_received(msg: Message) -> None:
     @callback
     def async_mqtt_connect(self):
         """Update state on connection/disconnection to MQTT broker."""
-        if self.hass.is_running:
```

Comment:
`hass.is_running` is true if state is `starting` or `running`.
However, during startup, state is `not_running` for quite some time. Is this intentional?

Reply:
Yes. We only return `is_running` when Home Assistant starts (fires the start event). We are not running during the setup phase.